cassandra/cassandra_test.go

```diff
@@ -0,0 +1,425 @@
+package cassandra
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/gocql/gocql"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	cassandracontainer "github.com/testcontainers/testcontainers-go/modules/cassandra"
+)
+
+const (
+	// cassandraImage is the default image used for running cassandra in tests.
+	cassandraImage              = "cassandra:latest"
+	cassandraImageEnvVar string = "TEST_CASSANDRA_IMAGE"
+	cassandraPort               = "9042/tcp"
+)
+
+// newTestStore creates a Cassandra container using the official module
+func newTestStore(t testing.TB) (*cassandracontainer.CassandraContainer, string, error) {
+	t.Helper()
+
+	img := cassandraImage
+	if imgFromEnv := os.Getenv(cassandraImageEnvVar); imgFromEnv != "" {
+		img = imgFromEnv
+	}
+
+	ctx := context.Background()
+
+	c, err := cassandracontainer.Run(ctx, img)
+	testcontainers.CleanupContainer(t, c)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Get connection parameters
+	host, err := c.Host(ctx)
+	if err != nil {
+		return nil, "", err
+	}
+
+	mappedPort, err := c.MappedPort(ctx, cassandraPort)
+	if err != nil {
+		return nil, "", err
+	}
+
+	connectionURL := host + ":" + mappedPort.Port()
+	return c, connectionURL, nil
+}
+
+// TestCassandraStorage tests the Cassandra storage implementation
+func TestCassandraStorage(t *testing.T) {
+	ctx := context.Background()
+
+	// Start Cassandra container
+	cassandraContainer, connectionURL, err := newTestStore(t)
+	if err != nil {
+		t.Fatalf("Failed to start Cassandra container: %v", err)
+	}
+	defer func() {
+		if err := cassandraContainer.Terminate(ctx); err != nil {
+			t.Logf("Failed to terminate container: %v", err)
+		}
+	}()
```

Comment:
bug: no need to defer the termination, as it was already done by the CleanupContainer call. More below, in every call to the `newTestStore` function.
```suggestion
```